hazelcast/src/test/java/com/hazelcast/internal/jmx/LocalStatsDelegateTest.java

```diff
@@ -16,168 +16,133 @@
-        Thread mapStatsThread2 = new Thread(new MapStatsThread(localStatsDelegate, done, 90));
-
-        mapStatsThread1.start();
-        mapStatsThread2.start();
-
-        Thread mapPutThread = new Thread(new MapPutThread(trial, done));
-        mapPutThread.start();
-
-        sleepSeconds(30);
-
-        done.set(true);
-
-        mapStatsThread1.join();
-        mapStatsThread2.join();
-        mapPutThread.join();
+    @Category(NightlyTest.class)
+    public void testMapStats_Created_OnlyOnce_InGivenInternal() throws Exception {
+        test(false, 60, 100, 90, 30);
     }
 
     /**
-     * In this test, map entry count is incrementing steadily. Thus, we expect OwnedEntryCount
-     * to increase as well. Since the interval value for generating new map stats is 5 seconds,
-     * we assert for greater or equal.
+     * In this test, map entry count is incrementing steadily. Thus, we expect OwnedEntryCount to increase as well. Since the
+     * interval value for generating new map stats is 5 seconds, we assert for greater or equal.
      */
     @Test
     @Category(NightlyTest.class)
-    public void stressTest() throws InterruptedException {
-        final IMap<Object, Object> trial = hz.getMap("trial");
+    public void stressTest() throws Exception {
+        test(true, 5, 100, 80, 60);
+    }
+
+    private void test(final boolean stress, final int intervalSec, final int mapStatsThread1SleepMs,
+            final int mapStatsThread2SleepMs, final int sleepSeconds) throws Exception {
+        final IMap<Object, Object> trial = createHazelcastInstance().getMap("trial");
 
         final StatsSupplier<LocalMapStats> statsSupplier = new LocalMapStatsSupplier(trial);
-        final LocalStatsDelegate localStatsDelegate = new LocalStatsDelegate<LocalMapStats>(statsSupplier, 5);
+        final LocalStatsDelegate<LocalMapStats> localStatsDelegate = new LocalStatsDelegate<>(statsSupplier, intervalSec);
 
-        Thread mapStatsThread1 = new Thread(new MapStatsThread(localStatsDelegate, done, true, 100));
-        Thread mapStatsThread2 = new Thread(new MapStatsThread(localStatsDelegate, done, true, 80));
+        final ExecutorService executor = Executors.newFixedThreadPool(3);
 
-        mapStatsThread1.start();
-        mapStatsThread2.start();
+        final Future<Void> mapStatsThread1 = executor
+                .submit(new MapStatsThread(localStatsDelegate, stress, mapStatsThread1SleepMs));
+        final Future<Void> mapStatsThread2 = executor
+                .submit(new MapStatsThread(localStatsDelegate, stress, mapStatsThread2SleepMs));
 
-        Thread mapPutThread = new Thread(new MapPutThread(trial, done));
+        final Thread mapPutThread = new Thread(new MapPutThread(trial));
         mapPutThread.start();
 
-        sleepSeconds(60);
+        sleepSeconds(sleepSeconds);
 
         done.set(true);
 
-        mapStatsThread1.join();
-        mapStatsThread2.join();
+        mapStatsThread1.get();
```

Comment:
alternative for waiting on both futures could be `FutureUtil#waitWithDeadline` utility method

Reply:
155f1eb8f8759485d056c3ca2867da7617dc073d